src/hyperlight_host/src/hypervisor/hyperlight_vm.rs

```diff
@@ -0,0 +1,1075 @@
+            regs[8] = vcpu_regs.r9; // r9
+            regs[9] = vcpu_regs.r8; // r8
+            regs[10] = vcpu_regs.rax; // rax
+            regs[11] = vcpu_regs.rcx; // rcx
+            regs[12] = vcpu_regs.rdx; // rdx
+            regs[13] = vcpu_regs.rsi; // rsi
+            regs[14] = vcpu_regs.rdi; // rdi
+            regs[15] = 0; // orig rax
+            regs[16] = vcpu_regs.rip; // rip
+            regs[17] = sregs.cs.selector as u64; // cs
+            regs[18] = vcpu_regs.rflags; // eflags
+            regs[19] = vcpu_regs.rsp; // rsp
+            regs[20] = sregs.ss.selector as u64; // ss
+            regs[21] = sregs.fs.base; // fs_base
+            regs[22] = sregs.gs.base; // gs_base
+            regs[23] = sregs.ds.selector as u64; // ds
+            regs[24] = sregs.es.selector as u64; // es
+            regs[25] = sregs.fs.selector as u64; // fs
+            regs[26] = sregs.gs.selector as u64; // gs
+
+            // Get the filename from the binary path
+            let filename = self.rt_cfg.binary_path.clone().and_then(|path| {
+                Path::new(&path)
+                    .file_name()
+                    .and_then(|name| name.to_os_string().into_string().ok())
+            });
+
+            // Include both initial sandbox regions and dynamically mapped regions
+            let mut regions: Vec<MemoryRegion> = self.sandbox_regions.clone();
+            regions.extend(self.mmap_regions.iter().map(|(_, r)| r).cloned());
+            Ok(Some(crashdump::CrashDumpContext::new(
+                regions,
+                regs,
+                xsave.to_vec(),
+                self.entrypoint,
+                self.rt_cfg.binary_path.clone(),
+                filename,
+            )))
+        } else {
+            Ok(None)
+        }
+    }
+}
+
+impl Drop for HyperlightVm {
+    fn drop(&mut self) {
+        self.interrupt_handle.set_dropped();
+    }
+}
+
+/// The vCPU tried to access the given addr
+enum MemoryAccess {
+    /// The accessed region has the given flags
+    AccessViolation(MemoryRegionFlags),
+    /// The accessed region is a stack guard page
+    StackGuardPageViolation,
+}
+
+/// Determines if a known memory access violation occurred at the given address with the given action type.
+/// Returns Some(reason) if violation reason could be determined, or None if violation occurred but in unmapped region.
+fn get_memory_access_violation<'a>(
+    gpa: usize,
+    tried: MemoryRegionFlags,
+    mut mem_regions: impl Iterator<Item = &'a MemoryRegion>,
+) -> Option<MemoryAccess> {
+    // find the region containing the given gpa
+    let region = mem_regions.find(|region| region.guest_region.contains(&gpa));
+
+    if let Some(region) = region {
+        if region.region_type == MemoryRegionType::GuardPage {
+            return Some(MemoryAccess::StackGuardPageViolation);
+        } else if !region.flags.contains(tried) {
+            return Some(MemoryAccess::AccessViolation(region.flags));
+        }
```

Comment:
Can we reach this point? is is possible to call this method, match a region, not be a guard page, and neither an access violation?
I assume this matches the previous behavior, if it does, and we are still not sure if this is reachable, consider adding a TODO comment to check if this is reachable